src/Microsoft.Health.Dicom.Core/Extensions/DicomDatasetExtensions.cs

```diff
@@ -329,5 +330,42 @@ public static void ValidateQueryTag(this DicomDataset dataset, QueryTag queryTag
                 minimumValidator.Validate(dicomElement);
             }
         }
+
+        /// <summary>
+        /// Gets DicomDatasets that matches a list of tags reprenting a tag path.
+        /// </summary>
+        /// <param name="dataset">The DicomDataset to be traversed.</param>
+        /// <param name="dicomTags">The Dicom tags modelling the path.</param>
+        /// <returns>Lists of DicomDataset that matches the list of tags.</returns>
+        public static IEnumerable<DicomDataset> GetSequencePathValues(this DicomDataset dataset, ReadOnlyCollection<DicomTag> dicomTags)
+        {
+            EnsureArg.IsNotNull(dataset, nameof(dataset));
+            EnsureArg.IsNotNull(dicomTags, nameof(dicomTags));
+
+            if (dicomTags.Count != 2)
+            {
+                throw new DicomValidationException(string.Join(",", dicomTags.Select(x => x.GetPath())), DicomVR.SQ, DicomCoreResource.NestedSequencesNotSupported);
+            }
+
+            var foundDatasets = new List<DicomDataset>();
```

Comment:
Instead of populating a list, should we just use `yield` here and below?

Reply:
Done